src/BaslerCamera.cpp

```diff
@@ -585,59 +478,66 @@ void Camera::_AcqThread::threadFunction()
-                            m_cam.StreamGrabber_->RetrieveResult(Result);
-                            if (Grabbed == Result.Status())
-                            {
-                                // Grabbing was successful, process image
-                                m_cam._setStatus(Camera::Readout,false);
-                                DEB_TRACE()  << "image#" << DEB_VAR1(m_cam.m_image_number) <<" acquired !";
 				if(!m_cam.m_video_flag_mode)
 				  {
-				    int nb_buffers;
-				    buffer_mgr.getNbBuffers(nb_buffers);
-				    if (m_cam.m_nb_frames == 0 || 
-					m_cam.m_image_number < int(m_cam.m_nb_frames - nb_buffers))
-				      m_cam.StreamGrabber_->QueueBuffer(Result.Handle(),NULL);
-                                
-				    HwFrameInfoType frame_info;
-				    frame_info.acq_frame_nb = m_cam.m_image_number;
-				    // copy TmpBuffer frame to SoftBuffer frame room
-				    if (m_cam.m_nb_frames == 0)
-				      {
-					void *ptr = buffer_mgr.getFrameBufferPtr(m_cam.m_image_number);
-					memcpy(ptr, (void *)Result.Buffer(), m_cam.ImageSize_);
-				      }
-				    continueAcq = buffer_mgr.newFrameReady(frame_info);
-				    DEB_TRACE() << DEB_VAR1(continueAcq);
-				  }
+
+        // Camera.StopGrabbing() is called automatically by the RetrieveResult() method
+        // when c_countOfImagesToGrab images have been retrieved.
+        while (m_cam.Camera_->IsGrabbing())
+        {
+            // Wait for an image and then retrieve it. A timeout of 3000 ms is used.
+            m_cam.Camera_->RetrieveResult(3000, ptrGrabResult, TimeoutHandling_ThrowException);
+
+            if (m_cam.m_status == Camera::Fault) {
+                m_cam._setStatus(Camera::Fault, false);
+                continueAcq = false;
+            }
+            // Image grabbed successfully?
+            if (ptrGrabResult->GrabSucceeded())
+            {
+                // Access the image data.
+                const uint8_t* pImageBuffer = (uint8_t*) ptrGrabResult->GetBuffer();
+                int nb_buffers;
+                buffer_mgr.getNbBuffers(nb_buffers);
+                            
+                HwFrameInfoType frame_info;
+                frame_info.acq_frame_nb = m_cam.m_image_number;
+                void *framePt = buffer_mgr.getFrameBufferPtr(m_cam.m_image_number);
+                const FrameDim& fDim = buffer_mgr.getFrameDim();
+                void* srcPt = ((char*)pImageBuffer);
+                DEB_TRACE() << "memcpy:" << DEB_VAR2(srcPt,framePt);
+                memcpy(framePt,srcPt,fDim.getMemSize());
+
+                continueAcq = buffer_mgr.newFrameReady(frame_info);
+                                            
+                DEB_TRACE() << DEB_VAR1(continueAcq);
+                ++m_cam.m_image_number;
+               
+                
+            #ifdef PYLON_WIN_BUILD
+                            // Display the grabbed image.
+                            Pylon::DisplayImage( 1, ptrGrabResult );
```

Comment:
I'm not sure a server should really display anything, even in a bizzare case of running it on windows.

Reply:
deleted